packages/plugin-pre/src/PreSeparatorHtmlDomRenderer.ts

```diff
@@ -0,0 +1,26 @@
+import { PreNode } from './PreNode';
+import { VNode } from '../../core/src/VNodes/VNode';
+import { AbstractRenderer } from '../../plugin-renderer/src/AbstractRenderer';
+import { HtmlDomRenderingEngine } from '../../plugin-html/src/HtmlDomRenderingEngine';
+
+export class PreSeparatorHtmlDomRenderer extends AbstractRenderer<Node[]> {
+    static id = HtmlDomRenderingEngine.id;
+    engine: HtmlDomRenderingEngine;
+
+    predicate = (item: VNode): boolean => {
+        const DefaultSeparator = this.engine.editor.configuration.defaults.Separator;
+        return item.is(DefaultSeparator) && !!item.ancestor(PreNode);
+    };
+
+    /**
+     * Render the VNode.
+     */
+    async render(node: VNode): Promise<Node[]> {
+        const separators = await this.super.render(node);
+        const rendering: Node[] = [];
+        for (let separatorIndex = 0; separatorIndex < separators.length; separatorIndex += 1) {
+            rendering.push(document.createTextNode('\n'));
+        }
```

Comment:
I'm not sure I understand. Isn't that the equivalent of:
```typescript
const rendering = separators.map(s => document.creatTextNode('\n'));
```

Reply:
It is but I figured you wouldn't like that because it would be using `map` for the side effect of rendering the nodes. :man_shrugging: